### ISSUE TITLE
BatchNormalization wrong training loss (correct validation loss) 

### ISSUE THREAD

**Issue by ozabluda** [keras-team/keras#8983]
Using same exact code and output as in https://github.com/keras-team/keras/issues/8982, we can see that validation loss is small:  `2.4932038522607058e-11` before training starts and `0.048399086749553677` after training is complete. However, training loss is large `0.6948` (and it's not the effect of averaging batches in an epoch).

I have no good hypothesis why it is happening but `0.6948` looks close to (but not quite) `1/sqrt(2)=0.7071`.

**Comment by ozabluda** [comment]
My best hypothesis for this behavior is:

During training, for `N=2`, "variance of the minibatch" is on average `1/2` of the population variance (`1.0`). So BN divides each sample by `stdev=1/sqrt(2)` (i.e. multiplies by `sqrt(2)`) "incorrect" `gamma=0.54530275`, is not used during training, so does not compensate for it. If so, we should expect `mse=1.0*(1-sqrt(2))^2)=0.17`. Not sure where  `0.6948` is coming from.

During inference,  "incorrect" `gamma=0.54530275` compensates for incorrect `moving_variance=0.47084832`

**Comment by ozabluda** [comment]
Never mind. Experimentally (see additions on the bottom of https://gist.github.com/ozabluda/84c36af4ced13f060524f743c7fe65c5) it checks out. Maybe factor of 0.5 is missing from my analytical answer, which I can't find.
